src/ui/render.ts

```diff
@@ -310,4 +329,88 @@ export default class UI {
 
     this.selectedHeaderCell = null;
   }
+
+  removeCellRangeSelection() {
+    const cells = Array.from(document.querySelectorAll("td"));
+    cells.forEach((cell) =>
+      cell.classList.remove("lightsheet_table_selected_cell_range"),
+    );
+  }
+
+  cellInRange(cell: HTMLTableCellElement) {
+    const { selectionStart, selectionEnd } = this.selectedCellsContainer;
+    if (!selectionStart || !selectionEnd) {
+      return false;
+    }
+
+    const { columnIndex: cellColumnIndex, rowIndex: cellRowIndex } =
+      this.getColumnAndRowIndex(cell);
+
+    if (cellColumnIndex === undefined || cellRowIndex === undefined)
+      return false;
+
+    const withinX =
+      (cellColumnIndex >= selectionStart.columnPosition &&
+        cellColumnIndex <= selectionEnd.columnPosition) ||
+      (cellColumnIndex <= selectionStart.columnPosition &&
+        cellColumnIndex >= selectionEnd.columnPosition);
+    const withinY =
+      (cellRowIndex >= selectionStart.rowPosition &&
+        cellRowIndex <= selectionEnd.rowPosition) ||
+      (cellRowIndex <= selectionStart.rowPosition &&
+        cellRowIndex >= selectionEnd.rowPosition);
+
+    return withinX && withinY;
+  }
+
+  updateSelection() {
+    this.removeCellRangeSelection();
+    const cells = Array.from(document.querySelectorAll("td"));
+    cells.forEach((cell) => {
+      if (
+        this.cellInRange(cell) &&
+        !cell.classList.contains("lightsheet_table_selected_cell")
+      ) {
+        cell.classList.add("lightsheet_table_selected_cell_range");
+      }
+    });
+  }
+
+  handleMouseDown(e: MouseEvent, cellDom: Element) {
+    if (e.button === 0) {
+      const { columnIndex: cellColumnIndex, rowIndex: cellRowIndex } =
+        this.getColumnAndRowIndex(cellDom);
+      this.selectedCellsContainer.selectionStart =
+        (cellColumnIndex != null || undefined) &&
+        (cellRowIndex != null || undefined)
+          ? {
+              rowPosition: Number(cellRowIndex),
+              columnPosition: Number(cellColumnIndex),
+            }
+          : null;
+    }
```

Comment:
If-block can be inverted here, as well as in `handleMouseUp`.

Reply:
This will be kept as it is since we will handle the right click in the next ticket. However, handleMouseUp(Now handleMouseOver) will be changed. Thanks.